src/app/services/data.service.ts

```diff
@@ -412,4 +413,37 @@ export class DataService {
   public getSetupCompleted(): boolean {
     return this.data.setupCompleted;
   }
+
+  public setFavoriteRoute(route: Route): void {
+    // this id will be unique and used as reference
+    // ToThink: What about having favorites A to B and B to A which are the same route but are not?
+    const newId = route.from.place_id + route.to.place_id;
+
+    const favRoute : FavoriteRoute = {
+      id: newId,
+      from: route.from,
+      to: route.to
+    } 
+
+
+    this.data.favoriteRoutes[newId] = favRoute;
+    this.setStorage();
+  }
+
+  public deleteFavoriteRoute(route: Route): void {
+    const deleteId = route.from.place_id + route.to.place_id;
+
+    delete this.data.favoriteRoutes[deleteId];
+    this.setStorage();
+  }
+
+  public isFavoriteRoute(route: Route): boolean {
```

Comment:
one could think about only placing this function in the route service, since it wont be used from the data service